server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1070,21 +1071,30 @@ private void cleanUpCompactors() {
       var groups = zoorw.getChildren(compactorQueuesPath);
 
       for (String group : groups) {
-        String qpath = compactorQueuesPath + "/" + group;
-
-        var compactors = zoorw.getChildren(qpath);
+        final String qpath = compactorQueuesPath + "/" + group;
+        final CompactorGroupId cgid = CompactorGroupId.of(group);
+        final var compactors = zoorw.getChildren(qpath);
 
         if (compactors.isEmpty()) {
           deleteEmpty(zoorw, qpath);
-        }
-
-        for (String compactor : compactors) {
-          String cpath = compactorQueuesPath + "/" + group + "/" + compactor;
-          var lockNodes = zoorw.getChildren(compactorQueuesPath + "/" + group + "/" + compactor);
-          if (lockNodes.isEmpty()) {
-            deleteEmpty(zoorw, cpath);
+          // Group has no compactors, we can clear its
+          // associated priority queue of jobs
+          getJobQueues().getQueue(cgid).clear();
```

Comment:
Its possible that a queue may not exists, so this could throw an NPE.

Reply:
In addition clearing the queue, we may want to set its size here to possibly lower it.  Maybe set it to  MANAGER_COMPACTION_SERVICE_PRIORITY_QUEUE_INITIAL_SIZE ?

Reply:
Handled NPE and set size to initial in 18d2243